### DISCUSSION TITLE
Possible to extract "street types" and "average speeds" in Valhalla?

### DISCUSSION THREAD

**Discussion by ghost** [valhalla/valhalla#3171] [Q&A]
I would like to know from a set of routes (e.g. A to B, C to D, ...) _what kind of streets_ ("Highway, Country Road, Streets in the City, ...) were used and what the expected _average velocity_ (according to the street sign) is.

Is this possible, using Valhalla?

Thank you for your reply!

**Comment by kevinkreiser** [comment]
yes it is possible. you can do this mostly with the OSRM format response but another way to get at this information is to do the route with the /route api and then call back again to the /trace_attributes api to select the attribution that you want. please note that "expected average velocity according to street sign" is not universally populated in OSM and will not be available on every edge in the graph. at any rate, please see the docs for the two apis listed above: https://github.com/valhalla/valhalla/blob/master/docs/api/turn-by-turn/api-reference.md and https://github.com/valhalla/valhalla/blob/master/docs/api/map-matching/api-reference.md#trace-attributes-action
